packages/block-editor/src/components/block-list/use-block-props/use-event-handlers.js

```diff
@@ -125,17 +120,23 @@ export function useEventHandlers( clientId ) {
 				event.preventDefault();
 			}
 
-			node.addEventListener( 'keydown', onKeyDown );
-			node.addEventListener( 'mouseleave', onMouseLeave );
-			node.addEventListener( 'dragstart', onDragStart );
+			node.addEventListener( 'focusin', onFocus );
+
+			if ( isSelected( clientId ) ) {
+				node.addEventListener( 'keydown', onKeyDown );
+				node.addEventListener( 'mouseleave', onMouseLeave );
+				node.addEventListener( 'dragstart', onDragStart );
+			}
 
 			return () => {
+				node.removeEventListener( 'focusin', onFocus );
 				node.removeEventListener( 'mouseleave', onMouseLeave );
 				node.removeEventListener( 'keydown', onKeyDown );
 				node.removeEventListener( 'dragstart', onDragStart );
 			};
 		},
 		[
+			clientId,
 			isSelected,
```

Comment:
One thing that changed, is this is now a function rather than the actual boolean value. Is it okay that we don't recompute this when the isSelected boolean value changes?

Reply:
Possibly some bad side effects given the failing e2e tests 😄